### PR TITLE
Create admin analytics api entry point

### PR DESCRIPTION
**Pull request by hervengenzi**
Added admin analytics controller and implemented answer and user relationships

### CODE REVIEW

File: app/controllers/api/admin/analytics_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Api
+  module Admin
```
Comment (bumi) [inline app/controllers/api/admin/analytics_controller.rb]: I don't have a good feeling with too much nesting. 
Can we move the Admin namespace out of the Api namespace? that makes it flatter and we have the user facing API and the Admin namespace separated.

File: app/models/question.rb [hunk 2]
```diff
@@ -19,4 +19,6 @@ def as_json(args = {})
       methods: %i[answers]
     ))
   end
+
+  delegate :count, to: :question, prefix: true
```
Comment (bumi) [inline app/models/question.rb]: thats' an error here I think?

**Comment by bumi** [comment]
I've added a note on the trello issue: https://trello.com/c/rA0jkHRT/30-create-admin-interface-with-evaluation-of-the-answers
I think we should first list the users and iterations and allow admins to see the result of a specific iteration.

File: app/controllers/api/admin/analytics_controller.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    class AnalyticsController < ActionController::API
+      def index
+        render json: Question.first.to_json(include: {
+
+                                              answers: {
+                                                methods: :user_count,
+                                              },
+
+                                            })
```
[the remark this answers is not in the document]
Comment (musayann) [reply app/controllers/api/admin/analytics_controller.rb]: I saw it on some rails boilerplate, but I didn't get to use it. but it seems straight forward

File: app/controllers/api/admin/analytics_controller.rb (hunk 1 above)
Comment (musayann) [reply app/controllers/api/admin/analytics_controller.rb]: But how can we differentiate the admin, Vue entry point from its API ??.
Or we just create an `admin_contoller` in the controllers' directory and then keep the admin API in the admin namespace only without the API one

File: app/models/question.rb (hunk 2 above)
Comment (musayann) [reply app/models/question.rb]: I think too, I don't see it's use

**Comment by bumi** [comment]
as discussed: 
let's try to use the same API endpoints for the admin section and use [pundit](https://github.com/varvet/pundit) for authorization.

**Comment by bumi** [comment]
I'd suggest we close this PR and add the rethought admin features in a new PR.

**Comment by musayann** [comment]
> I'd suggest we close this PR and add the rethought admin features in a new PR.

I agree too, we should start again with rethought features and less file modified

**Comment by fsmanuel** [comment]
tracking issue #43

**Pull request closed without merging**